src/FFmpegReader.cpp

```diff
@@ -926,11 +927,18 @@ std::shared_ptr<Frame> FFmpegReader::ReadStream(int64_t requested_frame) {
 	bool check_seek = false;
 	int packet_error = -1;
 
+	// Seeking can sometimes cause us to be quite far away from the requested
+	// frame, and thus, we need to iterate a large number of packets to find
+	// the requested position. This is mostly to prevent infinite loops when
+	// no more packets can be found though.
+	int attempts = 0;
+	int max_attempts = 30 * 30 * 60;
```

Comment:
Why `30 * 30 * 60` Should it be a number derived from the FPS?

Reply:
This is kind of a hack. In reality, we should be counting the packets read for `video` and `audio`, and keeping separate counts. Because, if we read a `video` packet, we need to keep calling `ProcessVideoPacket` until the counts match, for example, regardless of the `has_video` property.

Maybe this is a good time to fix that, because we should be able to delete this hacky code.

Reply:
Refactored this hack out of the code: https://github.com/OpenShot/libopenshot/pull/851